modules/drivers/mongo/src/metabase/driver/mongo.clj

```diff
@@ -183,9 +183,14 @@
 (defmethod driver/dbms-version :mongo
   [_ database]
   (with-mongo-connection [^com.mongodb.DB conn database]
-    (let [build-info (mg/command conn {:buildInfo 1})]
+    (let [build-info (mg/command conn {:buildInfo 1})
+          version-array (get build-info "versionArray")
+          sanitized-version-array (into [] (take-while nat-int?) version-array)]
+      (when (not= (take 3 version-array) (take 3 sanitized-version-array))
+        (log/warnf "sanitizing versionArray %s results in %s, losing information"
```

Comment:
how often is `dbms-version` called? Do we compute once and stick in the `metabase_database` table or get it dynamically a bunch? Wondering how many times this warn will appear.

Also, from the [thread](https://metaboat.slack.com/archives/C013N8XL286/p1684501559790059) the offending value was  `[4 2 18 -100]`. Your `sanitized-version-array` will throw away the `-100` but it's count will be 3 and you will not fire the warning that you are losing information despite throwing away the `-100`.

Reply:
The normal pattern is to get the version from the `:dbms_version` field of the DB object like this:

```clojure
(defmethod driver/database-supports? [:mongo :expressions]
  [_driver _feature db]
  (-> (:dbms_version db)
      :semantic-version
      (driver.u/semantic-version-gte [4 2])))
```

I don't intend to warn about losing anything after the patch version.